src/main/kotlin/org/usfirst/frc/team4099/robot/subsystems/Elevator.kt

```diff
@@ -0,0 +1,243 @@
+        PORTLOW(30.0), PORTMID(50.0), PORTHIGH(70.0),
+        VELOCITY_CONTROL(Double.NaN), OPEN_LOOP(Double.NaN)
+    }
+
+    enum class MovementState {
+        UP, DOWN, STILL
+    }
+
+    init {
+        talon.inverted = false
+        talon.clearStickyFaults(0)
+        talon.setSensorPhase(false)
+        talon.set(ControlMode.MotionMagic, 0.0)
+        talon.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, 0)
+        talon.configNominalOutputForward(0.0, 0)
+        talon.configNominalOutputReverse(0.0, 0)
+        talon.configPeakOutputReverse(-1.0, 0)
+        talon.configPeakOutputForward(1.0, 0)
+        talon.config_kP(0, Constants.Gains.ELEVATOR_UP_KP, 0)
+        talon.config_kI(0, Constants.Gains.ELEVATOR_UP_KI, 0)
+        talon.config_kD(0, Constants.Gains.ELEVATOR_UP_KD, 0)
+        talon.config_kF(0, Constants.Gains.ELEVATOR_UP_KF, 0)
+
+        talon.config_kP(1, Constants.Gains.ELEVATOR_DOWN_KP, 0)
+        talon.config_kI(1, Constants.Gains.ELEVATOR_DOWN_KI, 0)
+        talon.config_kD(1, Constants.Gains.ELEVATOR_DOWN_KD, 0)
+        talon.config_kF(1, Constants.Gains.ELEVATOR_DOWN_KF, 0)
+
+
+        talon.configMotionCruiseVelocity(0, 0)
+        talon.configMotionAcceleration(0, 0)
+
+        talon.configReverseSoftLimitEnable(true, 0)
+        talon.configReverseSoftLimitThreshold(-ElevatorConversion.inchesToPulses(70.0).toInt(), 0)
+        talon.overrideSoftLimitsEnable(true)
+
+        SmartDashboard.putNumber("elevator/pidPDown", Constants.Gains.ELEVATOR_DOWN_KP)
+        SmartDashboard.putNumber("elevator/pidIDown", Constants.Gains.ELEVATOR_DOWN_KI)
+        SmartDashboard.putNumber("elevator/pidDDown", Constants.Gains.ELEVATOR_DOWN_KD)
+        SmartDashboard.putNumber("elevator/pidFDown", Constants.Gains.ELEVATOR_DOWN_KF)
+
+        SmartDashboard.putNumber("elevator/pidPUP", Constants.Gains.ELEVATOR_UP_KP)
+        SmartDashboard.putNumber("elevator/pidIUP", Constants.Gains.ELEVATOR_UP_KI)
+        SmartDashboard.putNumber("elevator/pidDUP", Constants.Gains.ELEVATOR_UP_KD)
+        SmartDashboard.putNumber("elevator/pidFUP", Constants.Gains.ELEVATOR_UP_KF)
+    }
+
+    fun setOpenLoop(power: Double) {
+        elevatorState = ElevatorState.OPEN_LOOP
+        talon.set(ControlMode.PercentOutput, -power)
+    }
+
+
+    fun setElevatorVelocity(inchesPerSecond: Double) {
+        if ((inchesPerSecond <= 0 || Utils.around(inchesPerSecond, 0.0, .1)) && observedElevatorPosition < 0.5) {
+            setOpenLoop(0.0)
+            talon.sensorCollection.setQuadraturePosition(0, 0)
+//            println("exiting at 0 power, $inchesPerSecond")
+            return
+        }
+        elevatorState = ElevatorState.VELOCITY_CONTROL
+        if(inchesPerSecond >= 0) {
+            talon.selectProfileSlot(0, 0)
+        } else {
+            talon.selectProfileSlot(1, 0)
+        }
+        talon.set(ControlMode.Velocity, -inchesPerSecond)
```

Comment:
Is there a specific reason this is set to negative. If it should be negative, you should invert the talon rather than doing this.